pkgs/os-specific/linux/plymouth/default.nix

```diff
@@ -1,61 +1,81 @@
-{ stdenv, fetchurl, autoreconfHook, pkgconfig, libxslt, docbook_xsl
-, gtk3, udev, systemd
+{ stdenv, fetchFromGitLab
+, pkgconfig, autoreconfHook, libxslt, docbook_xsl
+, gtk3, udev, systemd, freetype, intltool
+, withLabelFt ? true
 }:
 
 stdenv.mkDerivation rec {
   pname = "plymouth";
-  version = "0.9.4";
+  version = "2020-03-25";
 
-  src = fetchurl {
-    url = "https://www.freedesktop.org/software/plymouth/releases/${pname}-${version}.tar.xz";
-    sha256 = "0l8kg7b2vfxgz9gnrn0v2w4jvysj2cirp0nxads5sy05397pl6aa";
+  src = fetchFromGitLab {
+    domain = "gitlab.freedesktop.org";
+    owner = "plymouth";
+    repo = "plymouth";
+    rev = "e31c81f9849c176d7b293ca79cc4507ba740c2fa";
+    sha256 = "sha256-kaiblXIinuiZ3vDzUQYOhYlyTnOGvpsLEOs8lQx9/Fo=";
   };
 
   nativeBuildInputs = [
-    autoreconfHook pkgconfig libxslt docbook_xsl
+    pkgconfig autoreconfHook
+    libxslt docbook_xsl
   ];
 
   buildInputs = [
-    gtk3 udev systemd
+    gtk3 udev systemd freetype intltool
   ];
 
-  postPatch = ''
-    sed -i \
-      -e "s#\$(\$PKG_CONFIG --variable=systemdsystemunitdir systemd)#$out/etc/systemd/system#g" \
-      -e "s#plymouthplugindir=.*#plymouthplugindir=/etc/plymouth/plugins/#" \
-      -e "s#plymouththemedir=.*#plymouththemedir=/etc/plymouth/themes#" \
-      -e "s#plymouthpolicydir=.*#plymouthpolicydir=/etc/plymouth/#" \
-      configure.ac
+  patches = [
+    ./fix-paths.patch
+
+    # https://build.opensuse.org/package/show/openSUSE%3AFactory/plymouth
+    ./manpages.patch
+    # ./only_use_fb_for_cirrus_bochs.patch
+  ] ++ stdenv.lib.optionals withLabelFt [
+    # https://gitlab.freedesktop.org/plymouth/plymouth/-/issues/45
+    ./v3-0001-ply-label-Don-t-crash-if-label-plugin-fails.patch
+    ./v3-0002-Add-label-ft-plugin.patch
+    ./v3-0004-Add-HiDPI-support-to-label-ft-plugin.patch
+  ];
+
+  postPatch = stdenv.lib.optionals withLabelFt ''
+    sed -i "s@/usr/share@/etc@" src/plugins/controls/label-ft/plugin.c
+    sed -i "s@/usr/bin/fc-match@fc-match@" src/plugins/controls/label-ft/plugin.c
   '';
 
+  configurePlatforms = [ "host" ];
+
   configureFlags = [
-    "--sysconfdir=/etc"
-    "--with-systemdunitdir=${placeholder "out"}/etc/systemd/system"
     "--localstatedir=/var"
+    "--with-runtimedir=/run"
+    "--with-systemdunitdir=${placeholder "out"}/etc/systemd/system"
+
     "--with-logo=/etc/plymouth/logo.png"
+    "--with-release-file=/etc/os-release"
     "--with-background-color=0x000000"
     "--with-background-start-color-stop=0x000000"
     "--with-background-end-color-stop=0x000000"
-    "--with-release-file=/etc/os-release"
-    "--without-system-root-install"
+
     "--without-rhgb-compat-link"
+    "--without-system-root-install"
+
+    "--enable-drm"
```

Comment:
Could you split clean ups and functional changes into a separate commit? It is hard to review otherwise.

Reply:
I think I split it up so that it's at least somewhat more readable? But at least the functionality improvements should be distinct now.